src/components/button/button.test.js

```diff
@@ -41,5 +41,128 @@ describe('/components/button', () => {
       const url = await page.url()
       expect(url).toBe(baseUrl + href)
     })
+    describe('preventing double clicks', () => {
+      it('does not block button links', async () => {
+        // Links should take a user to another part of the page or url, so debouncing it wouldnt have any impact.
+        // But we can check that we're not blocking any initial clicks.
+
+        await page.goto(baseUrl + '/components/button/link/preview', { waitUntil: 'load' })
```

Comment:
The way the logic is currently written, this test is never going to reach the part of the function that tests that the element is an `input` / `button`, because it'll already have returned because it doesn't have `data-prevent-double-click`.

What are we trying to test here? Do we need to be testing clicking a link _that has `data-prevent-double-click`_ set to `true`? Or is this test now redundant?